cpp/kiss_icp/core/Threshold.hpp

```diff
@@ -28,9 +28,10 @@ namespace kiss_icp {
 
 struct AdaptiveThreshold {
     explicit AdaptiveThreshold(double initial_threshold, double min_motion_th, double max_range)
-        : initial_threshold_(initial_threshold),
-          min_motion_th_(min_motion_th),
-          max_range_(max_range) {}
+        : min_motion_th_(min_motion_th),
+          max_range_(max_range),
+          model_error_sse2_(initial_threshold * initial_threshold),
```

Comment:
I like simplifying the code by doing this, but I think this is too hard to read/understand. Because we use the initial threshold to compute a hypothetical initial model error (without having computed a single pose, so what does a model error mean here?), such that the threshold matches the initial threshold we want.

However, I can not think of another solution right now that keeps this simplicity.

Reply:
Yeah I done that because in this way we not only have the initial threshold that we want but also the threshold is more robust to the initial _small motion_ issue that we have, for which, at the time, I design the `hasMoved` function. Cannot think about a better way as well, for me we can also hide this in the cpp if that make us feel more in peace.